internal/indexnode/task.go

```diff
@@ -346,7 +346,8 @@ func (it *IndexBuildTask) Execute(ctx context.Context) error {
 
 		getSavePathByKey := func(key string) string {
 			// TODO: fix me, use more reasonable method
-			return strconv.Itoa(int(it.req.IndexBuildID)) + "/" + strconv.Itoa(int(it.req.Version)) + "/" + strconv.Itoa(int(partitionID)) + "/" + strconv.Itoa(int(segmentID)) + "/" + key
+			return Params.IndexRootPath + "/" + strconv.Itoa(int(it.req.IndexBuildID)) + "/" + strconv.Itoa(int(it.req.Version)) + "/" +
+				strconv.Itoa(int(partitionID)) + "/" + strconv.Itoa(int(segmentID)) + "/" + key
```

Comment:
Maybe `path.Join` shall be used instead of manual concatenating?

Reply:
Use `path.Join` will nest many levels. Use ‘+’ maybe to be more intuitive？

Reply:
@xiaocai2333 this function signature is `func path.Join(elem ...string)`. All the elements can be joined in a single call.

Reply:
That would be great, I will do it.